packages/storybook/stories/va-pagination.stories.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import { getWebComponentDocs, propStructure } from './wc-helpers';
+import { VaPagination } from '@department-of-veterans-affairs/web-components/react-bindings';
+
+VaPagination.displayName = 'VaPagination';
+const paginationDocs = getWebComponentDocs('va-pagination');
+
+export default {
+  title: 'Components/va-pagination',
+};
+
+const defaultArgs = {
+  'disable-border': false,
+  'page': 3,
+  'pages': 5,
+  'show-last-page': false,
+};
+
+const Template = ({
+  'disable-border': disableBorder,
+  'page': currentPage,
+  pages,
+  'show-last-page': showLastPage,
+}) => {
+  const [page, setPage] = useState(currentPage);
+  const handlePageSelect = pageNumber => {
+    setPage(pageNumber);
+  };
+  return (
+    <VaPagination
+      disableBorder={disableBorder}
+      page={page}
+      pages={pages}
+      pageSelect={handlePageSelect}
```

Comment:
Seeing `pageSelect` prop, I'm not expecting that that's an event handler. Stencil [has ways](https://stenciljs.com/docs/events) of indicating that a component will fire a custom event, and that would let us have `onPageSelect` without any build warnings.

Reply:
I agree that it definitely feels wrong. The real problem is that the handlers for `onPageSelect` in `vets-website` don't handle events like handlers for `VaSelect` as an example.

Here is a handler for va-select used in vets-website. The handler receives the event and extracts what it needs from it.
```jsx
const onValueChange = option => {
    if (Object.values(FACILITY_SORT_METHODS).includes(option.detail.value)) {
      setSelectedSortMethod(option.detail.value);
      dispatch(updateCCProviderSortMethod(option.detail.value));
    } else {
      const selectedFacility = ccEnabledSystems.find(
        facility => facility.id === option.detail.value,
      );
      setSelectedSortMethod(selectedFacility.id);
      dispatch(
        updateCCProviderSortMethod(
          FACILITY_SORT_METHODS.distanceFromFacility,
          selectedFacility,
        ),
      );
    }
  };
```

Here is a handler for Pagination used in vets-website.
```jsx
  const onPageSelect = useCallback(
    selectedPage => {
      setPage(selectedPage); // setPage is a hook with an initial value of 1
      focusElement('#pagination-summary');
    },
    [setPage],
  );
```

Here are some options (not complete)
- We could update Pagination handlers in `vets-website` to handle events and update `onPageSelect` to be an event.
- We could rename Pagination's `onPageSelect` to `pageSelect` (currently in this PR).
- We could use `onPageSelect` as a prop and ignore the build warning.

@k80bowman @bkjohnson @pdavies88 @mendacitees for awareness/suggestions

Reply:
@andresriveratoro thanks for the thorough comments super helpful for awareness on why. I feel like we are in a catch-22 here and my gut says leave vets-website stuff alone because we could be opening a door here that doesn't need to be opened. Thoughts and feelings on if we remained the prop to pageSelectEvent? That may give someone that little bit more context and avoid the onXYZ error message situation and then add a second line in the comment above the prop explaining why we are doing this.

Reply:
I like the idea of using pageSelectEvent. I also like the idea of establishing a pattern with the web components for how we handle situations like this. IIRC something similar came up with another web component recently. Should we develop a standard way of handling these types of events?

Reply:
Something else to consider is how it will be displayed in Storybook. I like `pageSelectEvent`, but this will be displayed as a property and not an event in the autogenerated component-docs. I looked through our web components and wasn't able to find an example for function props except events.

Reply:
I [left a comment on the other thread](https://github.com/department-of-veterans-affairs/component-library/pull/275#discussion_r797878103) that's also relevant to this stuff, but I'm more interested in the first option. Based on that diff I put in the comment, that handler for the react Pagination component could be like this for the web component (I removed `useCallback` because I don't know what it's doing):

```js
// selectedPage is the custom event
const onPageSelect = selectedPage => {
    // selectedPage.detail is the data that we give the event - page number in this case
    setPage(selectedPage.detail); // setPage is a hook with an initial value of 1
    focusElement('#pagination-summary');
  },
  [setPage],
);
```

Reply:
Option 1 is fine with me if we can get it to work. I guess we could just add some Acceptance Criteria that says for the correlating vets-website PR that bumps versions of component-library package. Make sure all of the existing pagination handlers are updated to the correct format as well in that PR.

Reply:
There are 14 files that import the Pagination component, does this mean we would have to update the pagination handlers in all of those files?

Reply:
@k80bowman That's correct

Reply:
Or teams would be doing it when they do the migration.

Reply:
I wanted to elaborate a bit on my earlier comment about why I'm in favor of the event handler instead of a callback. A callback would get the job done, but callbacks aren't as robust as the browser's event model. This PR is an example, but I'll break it down a bit further: https://github.com/department-of-veterans-affairs/vets-website/pull/19716

We [originally](https://github.com/department-of-veterans-affairs/vets-website/blob/dcb56da4ff5b2ca649508f39280b7fb8b7a6e50e/src/platform/forms-system/src/js/components/SchemaForm.jsx#L92-L96) had a callback definition like this:

```jsx

onError() {
  const formContext = set('submitted', true, this.state.formContext);
  this.setState({ formContext });
  scrollToFirstError();
}
```

In https://github.com/department-of-veterans-affairs/vets-website/pull/19460 this got changed to parameterize the `submitted` value:

```jsx
onError(hasSubmitted = true) {
  const formContext = set('submitted', hasSubmitted, this.state.formContext);
  this.setState({ formContext });
  scrollToFirstError();
}
```

This _appeared_ fine. However, this callback was [already being called with an argument](https://github.com/department-of-veterans-affairs/react-jsonschema-form/blob/b10769a8c9ce1d69ac51febe8ad07af09cc12de2/src/components/Form.js#L101) that could be an empty array. If this callback had been an event handler this never would have happened. Instead, this caused a bug and led to a support issue.

Once we put a callback out there we become committed to supporting that definition, and if we want to make any big changes we can have a harder time of doing that depending on how the callback is being used by other teams. Relying on the event model decouples things a bit more and gives us more flexibility when it comes to future development. For example, we don't end up calling a callback that might have a signature like `errorCallback(arg1, arg2, somethingElse, data, thatOneFlagSomeoneWanted, options)`.

Reply:
@bkjohnson I appreciate you taking the time and digging through those older PRs to provide this context. Thanks!